apps/www/components/ui/toast.tsx

```diff
@@ -1,136 +1,113 @@
-import React from "react"
-import * as ToastPrimitive from "@radix-ui/react-toast"
-import { cva, type VariantProps } from "class-variance-authority"
+import * as React from "react"
+import * as ToastPrimitives from "@radix-ui/react-toast"
 import { X } from "lucide-react"
 
 import { cn } from "@/lib/utils"
+import { buttonVariants } from "./button"
 
-const ToastAction = React.forwardRef<
-  React.ElementRef<typeof ToastPrimitive.ToastAction>,
-  React.ComponentPropsWithoutRef<typeof ToastPrimitive.ToastAction>
+const ToastProvider = ToastPrimitives.Provider
+
+const ToastViewport = React.forwardRef<
+  React.ElementRef<typeof ToastPrimitives.Viewport>,
+  React.ComponentPropsWithoutRef<typeof ToastPrimitives.Viewport>
 >(({ className, ...props }, ref) => (
-  <ToastPrimitive.Action
+  <ToastPrimitives.Viewport
     ref={ref}
     className={cn(
-      "inline-flex h-6 items-center justify-center rounded-md border border-slate-400/30 bg-transparent px-2 text-sm font-medium leading-none transition-colors hover:bg-slate-500/10 focus:outline-none focus:ring-2 focus:ring-slate-400 focus:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=open]:bg-slate-100 dark:border-slate-50/50 dark:text-slate-100 dark:hover:bg-slate-100/30 dark:hover:text-slate-100 dark:focus:ring-slate-400 dark:focus:ring-offset-slate-900 dark:data-[state=open]:bg-slate-800",
+      "fixed bottom-6 right-6 z-50 max-h-screen w-full space-y-3 md:max-w-[400px]",
       className
     )}
     {...props}
   />
 ))
+ToastViewport.displayName = ToastPrimitives.Viewport.displayName
 
-ToastAction.displayName = ToastPrimitive.Action.displayName
+const Toast = React.forwardRef<
+  React.ElementRef<typeof ToastPrimitives.Root>,
+  React.ComponentPropsWithoutRef<typeof ToastPrimitives.Root>
+>(({ className, ...props }, ref) => {
+  return (
+    <ToastPrimitives.Root
+      ref={ref}
+      className={cn(
+        [
+          "data[swipe=cancel]:transition-transform grow-1 group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md border border-slate-200 bg-white p-6 pr-8 shadow-lg transition-all data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=open]:slide-in-from-bottom-full data-[state=closed]:slide-out-to-right-full dark:border-slate-700 dark:bg-slate-800",
```

Comment:
I would add something like `data-[swipe=move]:transition-none` or another workaround so the transition property is not applied while dragging, otherwise it feels laggy.

Reply:
Better yes !